tests/phpunit/tests/post/UpdateTermCountOnTransitionPostStatus.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * Tests for the _update_term_count_on_transition_post_status function.
+ *
+ * @group taxonomy
+ *
+ * @covers ::_update_term_count_on_transition_post_status
+ */
+class Tests_Taxonomy_UpdateTermCountOnTransitionPostStatus extends WP_UnitTestCase {
+
+	/**
+	 * @var int Post ID.
+	 */
+	protected $post_id;
+
+	/**
+	 * @var int Term ID.
+	 */
+	protected $term_id;
+
+	/**
+	 * @var string Post type.
+	 */
+	protected $post_type = 'post';
+
+	/**
+	 * @var string Taxonomy name.
+	 */
+	protected $taxonomy = 'category';
+
+	/**
+	 * Set up.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		register_post_type( $this->post_type, array( 'public' => true ) );
+		register_taxonomy( $this->taxonomy, $this->post_type, array( 'public' => true ) );
```

Comment:
To allow for compatibility with various versions of phpunit, [WP uses `set_up()` from the Yoast PHPunit Polyfills.](https://github.com/Yoast/PHPUnit-Polyfills?tab=readme-ov-file#option-1-yoastphpunitpolyfillstestcasestestcase)

```suggestion
	/**
	 * Set up.
	 */
	public function set_up(): void {
		parent::set_up();

		register_post_type( $this->post_type, array( 'public' => true ) );
		register_taxonomy( $this->taxonomy, $this->post_type, array( 'public' => true ) );
	}
```